src/main/java/fr/iut/pathpilotapi/routes/RouteService.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * RouteService.java                                  18 oct. 2024
+ * IUT de Rodez, no author rights
+ */
+
+package fr.iut.pathpilotapi.routes;
+
+import fr.iut.pathpilotapi.client.ClientService;
+import fr.iut.pathpilotapi.salesman.Salesman;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+/**
+ * service to manipulate routes
+ */
+@Service
+@RequiredArgsConstructor
+public class RouteService {
+
+    public static final String CLIENT_NOT_BELONGS_TO_SALESMAN = "The client does not belongs to the salesman";
+
+    private final RouteRepository routeRepository;
+    private final ClientService clientService;
+
+    /**
+     * Get all route from the database
+     * @return a list of all routes
+     */
+    public Page<Route> getAllRoutesFromSalesman(Pageable pageable, int salesmanId) {
+        return routeRepository.findAllBySalesman(salesmanId, pageable);
+    }
+
+    /**
+     * Get all route from the database owned by the salesman
+     * @param pageable
+     * @param salesman who owns the route
+     * @return a list of all routes
+     */
+    public Page<Route> getAllRoutesFromSalesman(Pageable pageable, Salesman salesman) {
+        return getAllRoutesFromSalesman(pageable, salesman.getId());
+    }
+
+    /**
+     * Create a new Route in the database.
+     * @param route    the Route to create
+     * @param salesman who creates the route
+     * @return the newly created Route
+     */
+    public Route addRoute(Route route, Salesman salesman) {
+
+        boolean isRouteValid = route.getClients_schedule().stream()
```

Comment:
Is this the only way to do it ?

Reply:
yes, but other solution would be way longer and less understandables